File: conway-java/src/main/java/com/gojek/pasti/conway/GameController.java
package com.gojek.pasti.conway;

import java.util.ArrayList;
import java.util.concurrent.TimeUnit;

public class GameController {
  private GameModel gameModel;
  private GamePresenter gamePresenter;

  public GameController(GameModel gameModel, GamePresenter gamePresenter) {
    this.gameModel = gameModel;
    this.gamePresenter = gamePresenter;
  }

  public void startGame(ArrayList<ArrayList<Character>> inputPattern, int iterations)
      throws InterruptedException {
    gameModel.populateGameBoard(inputPattern);
    int currentIteration = 0;

    while (currentIteration != iterations) {
      gamePresenter.printPattern();
      gameModel.nextIteration();
      TimeUnit.MILLISECONDS.sleep(100);
      currentIteration += 1;
    }
  }
}

File: conway-ruby/lib/conway/game_controller.rb
module Conway
  class GameController
    def initialize(game_model, game_presenter)
      @game_model = game_model
      @game_presenter = game_presenter
    end

    def start_game(input_pattern, iterations)
      @game_model.populate_game_board(input_pattern)
      @iterations = 0
      while @iterations != iterations
        @game_presenter.print_pattern
        @game_model.next_iteration
        sleep 0.05
        @iterations += 1
      end
    end
  end
end
